chain/client/src/test_utils.rs

```diff
@@ -1885,21 +1885,48 @@ impl TestEnv {
 
     pub fn process_partial_encoded_chunks(&mut self) {
         let network_adapters = self.network_adapters.clone();
-        for network_adapter in network_adapters {
-            // process partial encoded chunks
-            while let Some(request) = network_adapter.pop() {
-                if let PeerManagerMessageRequest::NetworkRequests(
-                    NetworkRequests::PartialEncodedChunkMessage {
-                        account_id,
-                        partial_encoded_chunk,
-                    },
-                ) = request
-                {
-                    self.shards_manager(&account_id).send(
-                        ShardsManagerRequestFromNetwork::ProcessPartialEncodedChunk(
-                            PartialEncodedChunk::from(partial_encoded_chunk),
-                        ),
-                    );
+
+        let mut keep_going = true;
+        while keep_going {
+            for network_adapter in network_adapters.iter() {
+                keep_going = false;
```

Comment:
@robin-near answering your comment from the main discussion here

In the now inner loop processing of a PEC may lead to PECF send to clients that were already processed and those messages would get lost. This is why I added the outer loop.

Reply:
Didn't completely understand what's going on here. Ignoring for now.

Reply:
old implementation only handled the PartialEncodedChunkMessage (PEC)
new implementation also handles PartialEncodedChunkForward (PECF)
new implementation keeps looping as long as there are any messages processed at all in the previous iteration - because receiving a PEC triggers sending PECF and we want to make sure we handle all of the messages